packages/ui/src/components/shared/menu/ImportAccountDialog/TransferAccountBackup.tsx

```diff
@@ -52,34 +53,45 @@ function TransferAccountBackup({ backup, resetBackup, onClose }: TransferAccount
       </div>
     );
 
-  const doImportAccount = (event: FormEvent) => {
+  const doVerifyAccountBackupPassword = async (event: FormEvent) => {
     event.preventDefault();
 
-    setOnImporting(true);
+    try {
+      await keyring.verifyAccountBackupPassword(backup, password);
+      setAccountBackupPasswordVerified(true);
+    } catch (e: any) {
+      toast.error(t<string>(e.message));
+    }
+  };
 
+  const doImportAccount = (walletPassword: string) => {
     setTimeout(async () => {
```

Comment:
We can probably remove this `setTimeout` since the `onPasswordVerified` is already running inside a timeout.